gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/MemberLogsMessage.java

```diff
@@ -0,0 +1,392 @@
+import com.gemstone.gemfire.distributed.DistributedMember;
+import com.gemstone.gemfire.distributed.internal.InternalDistributedSystem;
+import com.gemstone.gemfire.internal.cache.GemFireCacheImpl;
+import com.gemstone.gemfire.internal.lang.StringUtils;
+import com.google.common.io.ByteStreams;
+import com.pivotal.gemfirexd.internal.engine.distributed.message.GfxdFunctionMessage;
+import com.pivotal.gemfirexd.internal.engine.distributed.message.MemberExecutorMessage;
+import org.apache.log4j.Appender;
+import org.apache.log4j.FileAppender;
+import org.apache.log4j.Logger;
+
+
+public class MemberLogsMessage extends MemberExecutorMessage {
+
+  private Logger logger = Logger.getLogger(MemberLogsMessage.class);
+
+  private String memberId;
+  private Long offset;
+  //private int byteLength;
+  private Long byteLength;
+
+  /** Default constructor for deserialization. Not to be invoked directly. */
+  public MemberLogsMessage() {
+    super(true);
+  }
+
+  public MemberLogsMessage(
+      final ResultCollector<Object, Object> rc) {
+    super(rc, null, false, true);
+  }
+
+  private MemberLogsMessage(MemberExecutorMessage other) {
+    super(other);
+  }
+
+  public String getMemberId() {
+    return memberId;
+  }
+
+  public void setMemberId(String memberId) {
+    this.memberId = memberId;
+  }
+
+  public Long getOffset() {
+    return offset;
+  }
+
+  public void setOffset(Long offset) {
+    this.offset = offset;
+  }
+
+  public Long getByteLength() {
+    return byteLength;
+  }
+
+  public void setByteLength(Long byteLength) {
+    this.byteLength = byteLength;
+  }
+
+  @Override
+  protected void execute() throws Exception {
+
+    GemFireCacheImpl gemFireCache = GemFireCacheImpl.getExisting();
+    InternalDistributedSystem ids = gemFireCache.getDistributedSystem();
+    String memberId = ids.getMemberId();
+
+    Map logDetailsMap = new HashMap();
+    logDetailsMap.put("id", memberId);
+    logDetailsMap.put("name", ids.getName());
+    logDetailsMap.put("userDir", getUserDir());
+    logDetailsMap.put("logFile", getLogFile());
+    logDetailsMap.put("logData", getLog(this.offset, this.byteLength));
+
+    lastResult(logDetailsMap);
+
+  }
+
+  private String getUserDir(){
+    return System.getProperty("user.dir");
+  }
+
+  private String getLogFile() {
+    Logger rootLogger = Logger.getRootLogger();
+    Appender appender;
+    if (rootLogger != null) {
+      Enumeration<?> e = rootLogger.getAllAppenders();
+      while (e.hasMoreElements()) {
+        appender = (Appender)e.nextElement();
+        if (appender instanceof FileAppender) {
+          return ((FileAppender)appender).getFile();
+        }
+      }
+    }
+    return "";
+  }
+
+  /** Get the part of the log files given the offset and desired length of bytes */
+  private Map<String, Object> getLog(Long offset, Long byteLength) {
+
+    Map<String, Object> logsMap = new HashMap<String, Object>();
+
+    String logFile = getLogFile();
```

Comment:
You could pass this from execute() which has already determined this instead of determining again.